chalicelib/checks/wfr_checks.py

```diff
@@ -1,1448 +1,1456 @@
-    check.summary = ""
-    check.full_output = {}
-    check.status = 'PASS'
-
-    # check indexing queue
-    env = connection.ff_env
-    indexing_queue = ff_utils.stuff_in_queues(env, check_secondary=True)
-
-    if indexing_queue:
-        check.status = 'PASS'  # maybe use warn?
-        check.brief_output = ['Waiting for indexing queue to clear']
-        check.summary = 'Waiting for indexing queue to clear'
-        check.full_output = {}
-        return check
-
-    # start with cases with a metawfr and no ingested final vcf
-    query = '/search/?type=Case&meta_workflow_run!=No+value&vcf_file=No+value'
-    search_res = ff_utils.search_metadata(query, key=my_auth)
-
-    # filter those whose samples do not have processed_files
-    filtered_res = []
-    for r in search_res:
-        SNV_processed = 0
-        SV_processed = 0
-        result_list = r['sample_processing'].get('processed_files', [])
-        for pf in result_list:
-            try:
-                pf['variant_type']
-                if pf['variant_type'] == "SV":
-                    SV_processed += 1
-                elif pf['variant_type'] == "SNV":
-                    SNV_processed += 1
-            except:
-                SNV_processed += 1
-        if SNV_processed < 2:
-            filtered_res.append(r)
-
-    # nothing to run
-    if not filtered_res:
-        check.summary = 'All Good!'
-        return check
-
-    metawfr_uuids = [r['meta_workflow_run']['uuid'] for r in filtered_res]
-    metawfr_titles = [r['meta_workflow_run']['display_title'] for r in filtered_res]
+    Check for finding full annotated VCFs that have failed ingestion, so that they
+    can be reset and the ingestion rerun if needed.
+    """
+    check = initialize_check(connection)
+    check.action = "reset_vcf_ingestion_errors"
+    check.description = (
+        "Find VCFs that have failed ingestion to clear metadata for reingestion"
+    )
 
-    check.allow_action = True
-    check.summary = 'Some metawfrs may need patching sample processing.'
-    check.status = 'WARN'
-    msg = str(len(metawfr_uuids)) + ' metawfrs may need patching sample processing'
+    files_with_ingestion_errors = {}
+    accessions = format_kwarg_list(file_accessions)
+    ingestion_error_search = "search/?type=FileProcessed&file_ingestion_status=Error"
+    if accessions:
+        ingestion_error_search += "&accession="
+        ingestion_error_search += "&accession=".join(accessions)
+    ingestion_error_search += "&field=@id&field=file_ingestion_error"
+    search_response = ff_utils.search_metadata(
+        ingestion_error_search, key=connection.ff_keys
+    )
+    for result in search_response:
+        file_atid = result.get("@id")
+        first_error = None
+        ingestion_errors = result.get("file_ingestion_error", [])
+        if ingestion_errors:
+            # usually there are 100 errors, but just report first error here
+            first_error = ingestion_errors[0].get("body")
```

Comment:
There are 100 errors just because that is the number I chose to truncate at and usually if there is an error it presents itself on many rows. It is not always the case though. I would consider reporting at least 5-10 of the errors, that way we should always get enough info on the class of error from the check (and not have to check portal metadata additionally).